front/types/assistant/configuration.ts

```diff
@@ -0,0 +1,66 @@
+import { ModelId } from "@app/lib/databases";
+import {
+  RetrievalQuery,
+  RetrievalTimeframe,
+} from "@app/types/assistant/actions/retrieval";
+
+/**
+ * Agent config
+ */
+export type AgentConfigurationStatus = "active" | "archived";
+export type AgentConfigurationScope = "global" | "workspace";
+export type AgentConfigurationType = {
+  sId: string;
+  status: AgentConfigurationStatus;
+  name: string;
+  pictureUrl: string | null;
+  action: AgentActionConfigurationType | null; // If undefined, no action performed
+  generation: AgentGenerationConfigurationType | null; // If undefined, no text generation.
+};
+
+/**
+ * Generation config
+ */
+export type AgentGenerationConfigurationType = {
+  id: ModelId;
+  prompt: string;
+  model: {
+    providerId: string;
+    modelId: string;
+  };
+};
+
+/**
+ * Action > Retrieval
+ */
+export type AgentActionConfigurationType = RetrievalConfigurationType;
+
+/**
```

Comment:
As per IRL Let's keep all that in `retrieval.ts` because really it's retrieval related.